### PR TITLE
build: drop os and use pathlib in generate-examples.py

### LINKED ISSUE #1213: build: generate images – python  drop os and use pathlib
Could we drop os and instead use 
```suggestion
from pathlib import Path
```

Makes everything easier and more readable.
We don't use `with open` and don't have to go over all the files if we use `glob`.

_Originally posted by @rettinghaus in https://github.com/music-encoding/music-encoding/pull/1184#discussion_r1211217227_

### PR DESCRIPTION
**Pull request by rettinghaus**
This PR updates the Python script used for example generation by replacing the os module with the more modern path lib module.

closes #1213

### CODE REVIEW

File: utils/examples/generate-examples.py [hunk 1]
```diff
@@ -52,32 +52,25 @@
     logging.basicConfig(format="[%(asctime)s] [%(levelname)8s] %(message)s (%(filename)s:%(lineno)s)",
                         level=level)
     log = logging.getLogger(__name__)
-    log.info("Running at logging level %s", level_msg)
+    log.info(f"Running at logging level {level_msg}")
```
Comment (ahankinson) [inline utils/examples/generate-examples.py]: Using f-strings always computes the value, while printf-style defers computing it if the message wouldn't be printed. 

See: https://docs.python.org/3/howto/logging.html#optimization

File: utils/examples/generate-examples.py [hunk 2]
```diff
@@ -62,22 +62,15 @@
...
             continue
 
         # Download the MEI file from the given url
-        mei_example: str = ""
-        with open(mei_file) as f:
-            mei_example = f.read()
+        mei_example: str = mei_file.read_text()
 
         if not tk.loadData(mei_example):
             log.error("Failed to load %s", mei_file)
```
Comment (musicEnfanthen) [inline utils/examples/generate-examples.py]:
```suggestion
            log.error("Failed to load %s", mei_file.name)
```

**Review by musicEnfanthen — COMMENTED** [review]
Just a small addition, otherwise looks good to me

**Review by musicEnfanthen — APPROVED** [review]

**Review by bwbohl — APPROVED** [review]

**Comment by musicEnfanthen** [comment]
Thank you!

**Pull request merged**